astropy/cosmology/parameter/_typing.py

```diff
@@ -0,0 +1,13 @@
+"""Type hints for the parameter module."""
+
+__all__: list[str] = []
+
+from collections.abc import Callable
+from typing import Any, TypeAlias, TypeVar
+
+from astropy.cosmology.core import Cosmology
+
+from ._core import Parameter
+
+VT = TypeVar("VT")
+_FValidateCallable: TypeAlias = Callable[[Cosmology, Parameter, Any], VT]  # noqa: PYI047
```

Comment:
Does this name still need an underscore? The fact that this is in `_typing` and it is not listed in `__all__` should be enough to declare it as private.
```suggestion
FValidateCallable: TypeAlias = Callable[[Cosmology, Parameter, Any], VT]

Reply:
Let's see if this breaks RTD!

Reply:
It breaks RTD

Reply:
@eerovaher, fine to revert?

Reply:
Sphinx complaining must mean that there public signatures that refer to `FValidateCallable`, so the alias must be made public too. Otherwise we are telling users that something must be an `FValidateCallable` without telling them what that means.